system/HTTP/ResponseTrait.php

```diff
@@ -567,8 +568,17 @@ public function setCookie(
             return $this;
         }
 
+        /** @var CookieConfig|null $cookieConfig */
+        $cookieConfig = config('Cookie');
+
+        if ($cookieConfig) {
```

Comment:
using instanceof is better  I think, that's we know that type is exactly a CookieConfig:

```suggestion
        if ($cookieConfig instanceof CookieConfig) {
```